chain/xrp/client/client.go

```diff
@@ -254,10 +262,139 @@ type DeletedNode struct {
 	LedgerIndex     string       `json:"LedgerIndex,omitempty"`
 }
 
+type CreatedNodeWrapper struct {
+	node *CreatedNode
+}
+
+func (mnw *CreatedNodeWrapper) IsValidMovement() bool {
+	return mnw.node.LedgerEntryType == "AccountRoot" || mnw.node.LedgerEntryType == "RippleState"
+}
+
+func (mnw *CreatedNodeWrapper) GetAddress(txResponse *TransactionResponse) (xc.Address, error) {
+	if mnw.node.NewFields == nil {
+		return "", fmt.Errorf("empty NewFields in CreatedNode")
+	}
+
+	if mnw.node.LedgerEntryType == "AccountRoot" {
+		return xc.Address(mnw.node.NewFields.Account), nil
+
+	} else if mnw.node.LedgerEntryType == "RippleState" {
+		isSource, fetchIsSourceErr := mnw.IsSource(txResponse)
+		if fetchIsSourceErr != nil {
+			return "", fetchIsSourceErr
+		}
+
+		if isSource {
+			if mnw.node.NewFields.HighLimit == nil {
+				return "", fmt.Errorf("empty HighLimit in NewFields")
+			}
+
+			return xc.Address(mnw.node.NewFields.HighLimit.Issuer), nil
+		} else {
+			if mnw.node.NewFields.LowLimit == nil {
+				return "", fmt.Errorf("empty HighLimit in NewFields")
+			}
+
+			return xc.Address(mnw.node.NewFields.LowLimit.Issuer), nil
+		}
+	}
+
+	return "", fmt.Errorf("unknown node type in CreatedNode")
+}
+
+func (mnw *CreatedNodeWrapper) GetContract(txResponse *TransactionResponse) (xc.ContractAddress, error) {
+	var (
+		contract                  xc.ContractAddress
+		finalBalanceHumanReadable xc.AmountHumanReadable
+		err                       error
+	)
+
+	if mnw.node.NewFields == nil {
+		return "", fmt.Errorf("empty NewFields in CreatedNode")
+	}
+
+	if mnw.node.NewFields.Balance.XRPAmount != "" {
+		contract = ""
+	} else {
+		finalBalanceHumanReadable, err = xc.NewAmountHumanReadableFromStr(mnw.node.NewFields.Balance.TokenAmount.Value)
+		if err != nil {
+			return "", err
+		}
+
+		finalBalanceBlockchain := finalBalanceHumanReadable.ToBlockchain(6)
+		zero := xc.NewAmountBlockchainFromUint64(0)
+
+		if finalBalanceBlockchain.Cmp(&zero) > 0 {
+			if mnw.node.NewFields.HighLimit == nil {
+				return "", fmt.Errorf("empty HighLimit in NewFields")
+			}
+
+			contract = xc.ContractAddress(mnw.node.NewFields.HighLimit.Issuer)
+		} else {
+			if mnw.node.NewFields.LowLimit == nil {
+				return "", fmt.Errorf("empty LowLimit in NewFields")
+			}
+
+			contract = xc.ContractAddress(mnw.node.NewFields.LowLimit.Issuer)
+		}
+	}
+
+	return contract, nil
```

Comment:
A contract should be something like `USD-ABCD...` for it to be useful in other endpoints.  Need to prepend the symbol.